src/mlpack/core/data/one_hot_encoding.hpp

```diff
@@ -28,8 +28,21 @@ namespace data {
  * @param labelsIn Input labels of arbitrary datatype.
  * @param output Binary matrix.
  */
-template<typename eT, typename RowType>
+template<class RowType, template <typename> class MatType, class eT>
 void OneHotEncoding(const RowType& labelsIn,
+                    MatType<eT>& output);
+
+/**
+ * Overloaded function for the above function, which takes a matrix as input
+ * and also a vector of indices to encode and outputs a matrix.
```

Comment:
Can you document what the vector of indices means?  I think you mean that the indices in `indices` represent the IDs of the dimensions to be one-hot encoded?  (If so, shouldn't it be, e.g., `arma::Col<size_t>`?)

Reply:
Done